src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+            Account a = new Account(Id = i, Primary_Contact__c = updateAccPrimaryContact.get(i));
+            //Inject the Adv logic to handle npsp primary contact of the Account
+            if (Advancement_Info.useAdv() && Advancement_Info.getApiHEDA() != NULL) {
+                Advancement_Info.getApiHEDA().primaryContact(a, updateAccPrimaryContact.get(i));
             }
+            accountsToUpdate.add(a);
+        }
 
-            list<Account> listAcc = [Select Id, RecordTypeId, Primary_Contact__c, OwnerId, (select Id from Contacts limit 2) from Account
-                where Id IN :setAccId];
-
-            for (Account acc : listAcc) {
-                // only consider HH and Admin Accounts
-                //TODO: use setting to determine if Account owner should be changed (W-009780)
-                if (acc.RecordTypeId != null
-                        && (acc.RecordTypeId == userDefinedHHRecordTypeId
-                            || acc.RecordTypeId == userDefinedAdminRecordTypeId)) {
-                    // if there is only 1 contact associated with this account
-                    if (acc.Contacts.size() == 1 && acc.Primary_Contact__c != null) {
-                    	Contact c = mapContactIdContactOwnerChange.get(acc.Primary_Contact__c);
-                    	Id ownerIdContact = (c != null ? c.OwnerId : null);
-                    	if (acc.OwnerId != ownerIdContact) {
-                    		acc.OwnerId = ownerIdContact;
-                    		listAccToUpdate.add(acc);
-                    	}
-                    }
+        if (accountsToUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)accountsToUpdate);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Contact's owner has changed. Consider updating related account owner.
+    * @param contactOwnerChange a map of Contact Id to Contact, for the contacts to consider.
+    * @param dmlWrapper to hold the Contacts that need update.
+    */
+    private void updateOwners(Map<Id,Contact> contactOwnerChange, DmlWrapper dmlWrapper) {
+        List<Account> accountsToUpdate = new List<Account>();
+        Set<Id> contactsAccountId = new Set<Id>();
+
+        if (contactOwnerChange.size() == 0) {
+            return;
+        }
+
+        for (Contact con : contactOwnerChange.values()) {
+            contactsAccountId.add(con.AccountId);
+        }
+
+        List<Account> listAcc = [SELECT Id, RecordTypeId, Primary_Contact__c, OwnerId,
+                                (SELECT Id FROM Contacts limit 2)
+                                 FROM Account
+                                 WHERE Id IN :contactsAccountId];
+
+        for (Account acc : listAcc) {
+            //TODO: use setting to determine if Account owner should be changed (W-009780)
+            if (acc.Contacts.size() == 1 && acc.Primary_Contact__c != NULL &&
```

Comment:
Seems like we have a number of comparisons here as well. any chance we can make them more readable? for example, for me it is very easy for me to miss that last grouping. I like creating some booleans and passing those in instead since it allows you to be more explicit with the variable name, and a bit easier to read what conditions you're testing.

Reply:
@lifewithryan , sure will make the update for this.